### PR TITLE
Nml 2127 - Update PoET simulator and Validator Registry Transaction Processor to Support Enclave Quote Structure

### PR DESCRIPTION
**Pull request by jjason**
Updates the PoET simulator and validator registry transaction processor:

* Python wrapper classes created for supporting the enlcave quote and embedded structures found in IAS attestation verification report. Classes support serializing/deserializing between Python objects and binary layout of corresponding C/C++ structs.
* Simulator uses SGX structs wrapper to create enclave quote layout the same as used by SGX enclave implementation
* Simulator and validator registry transaction processor verify enclave quote in attestation verification report in the same was as SGX enclave implementation
* Validator registry transaction processor unit tests updated for new enclave quote layout

### CODE REVIEW

File: consensus/poet/families/sawtooth_validator_registry/validator_registry/processor/handler.py [hunk 1]
```diff
@@ -121,6 +122,17 @@ def _set_data(state, address, data):
 
 class ValidatorRegistryTransactionHandler(object):
 
+    # The basename and enclave measurement values we expect to find
+    # in the enclave quote in the attestation verification report.
+    __VALID_BASENAME__ = \
```
Comment (jsmitchell) [inline consensus/poet/families/sawtooth_validator_registry/validator_registry/processor/handler.py]: Where do BASENAME and ENCLAVE_MEASUREMENT come from? Can they change? Can you give some commentary on hardcoding?

File: consensus/poet/common/sawtooth_poet_common/sgx_structs/_sgx_attributes.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#
+# INTEL CONFIDENTIAL
```
Comment (peterschwarz) [inline consensus/poet/common/sawtooth_poet_common/sgx_structs/_sgx_attributes.py]: This is not the same license we are using in the rest of the project - needs to be the Apache 2.0 license

File: consensus/poet/common/sawtooth_poet_common/sgx_structs/_sgx_basename.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
+#
```
Comment (peterschwarz) [inline consensus/poet/common/sawtooth_poet_common/sgx_structs/_sgx_basename.py]: Fix license in all locations.

File: consensus/poet/common/sawtooth_poet_common/sgx_structs/_sgx_attributes.py (hunk 2 above)
Comment (jjason) [inline consensus/poet/common/sawtooth_poet_common/sgx_structs/_sgx_attributes.py]: Crud.  Will do.  Stupid copy/paste error as I took original file from sawtooth-poet repo.

File: consensus/poet/families/sawtooth_validator_registry/validator_registry/processor/handler.py (hunk 1 above)
Comment (jjason) [inline consensus/poet/families/sawtooth_validator_registry/validator_registry/processor/handler.py]: These are actual values from an IAS verification report generated by SGX enclave.  They are currently placeholders (see comments in code when they are checked in simulator or validator registry transaction processor).  I think that the intent is going to be that these values have to somehow be configured for the validators - I don't now if the config transaction family is the place to put these values, but we are probably going to have some small set  of values that are valid as I believe that the enclave measurement will be different whenever we update the SGX PoET enclave.

**Comment by jjason** [comment]
Updated copyrights and pushed new PR.

**Pull request merged**